src/merkle_tree.rs

```diff
@@ -0,0 +1,291 @@
+use std::vec;
+use hex;
+use sha2::{Digest, Sha256};
+use crate::merkle_error::MerkleError;
+use crate::proof_element::ProofElement;
+
+#[derive(Debug, PartialEq)]
+pub struct MerkleTree{
+    tree: Vec<Vec<String>>
```

Comment:
I would create a type Hash in order to make the code a little bit more clear, as I think it looks better than handling Strings all the time. And with this for example you could make a distinction between `MerkleTree::build` and `MerkleTree::build_without_hashing` as one would receive strings and the other hashes.

Reply:
Done!